### ISSUE TITLE
New parser: postconf -M

### ISSUE THREAD

**Issue by chriscroome** [kellyjonbrazil/jc#248]
The default output from the Postfix configuration utility, [postconf](https://manpages.debian.org/testing/postfix/postconf.1.en.html), works very well with the ini parser, for example:

```bash
postconf | jc --ini -p
```

However I haven't found a way to easily load the output when run with [-M](https://manpages.debian.org/testing/postfix/postconf.1.en.html#M), *"Show master.cf file contents instead of main.cf file contents"*, for example:

```
postconf -M      
smtp       inet  n       -       y       -       -       smtpd
pickup     unix  n       -       y       60      1       pickup
cleanup    unix  n       -       y       -       0       cleanup
qmgr       unix  n       -       n       300     1       qmgr
tlsmgr     unix  -       -       y       1000?   1       tlsmgr
rewrite    unix  -       -       y       -       -       trivial-rewrite
bounce     unix  -       -       y       -       0       bounce
defer      unix  -       -       y       -       0       bounce
trace      unix  -       -       y       -       0       bounce
verify     unix  -       -       y       -       1       verify
flush      unix  n       -       y       1000?   0       flush
proxymap   unix  -       -       n       -       -       proxymap
proxywrite unix  -       -       n       -       1       proxymap
smtp       unix  -       -       y       -       -       smtp
relay      unix  -       -       y       -       -       smtp -o syslog_name=postfix/$service_name
showq      unix  n       -       y       -       -       showq
error      unix  -       -       y       -       -       error
retry      unix  -       -       y       -       -       error
discard    unix  -       -       y       -       -       discard
local      unix  -       n       n       -       -       local
virtual    unix  -       n       n       -       -       virtual
lmtp       unix  -       -       y       -       -       lmtp
anvil      unix  -       -       y       -       1       anvil
scache     unix  -       -       y       -       1       scache
postlog    unix-dgram n  -       n       -       1       postlogd
maildrop   unix  -       n       n       -       -       pipe flags=DRXhu user=vmail argv=/usr/bin/maildrop -d ${recipient}
uucp       unix  -       n       n       -       -       pipe flags=Fqhu user=uucp argv=uux -r -n -z -a$sender - $nexthop!rmail ($recipient)
ifmail     unix  -       n       n       -       -       pipe flags=F user=ftn argv=/usr/lib/ifmail/ifmail -r $nexthop ($recipient)
bsmtp      unix  -       n       n       -       -       pipe flags=Fq. user=bsmtp argv=/usr/lib/bsmtp/bsmtp -t$nexthop -f$sender $recipient
scalemail-backend unix - n       n       -       2       pipe flags=R user=scalemail argv=/usr/lib/scalemail/bin/scalemail-store ${nexthop} ${user} ${extension}
mailman    unix  -       n       n       -       -       pipe flags=FRX user=list argv=/usr/lib/mailman/bin/postfix-to-mailman.py ${nexthop} ${user}
```
The columns are whitespace separated apart from the last one, the names of the columns are:

```
# ==========================================================================
# service type  private unpriv  chroot  wakeup  maxproc command + args
#               (yes)   (yes)   (no)    (never) (100)
# ==========================================================================
```
See the [syntax](https://manpages.debian.org/testing/postfix/master.5.en.html#SYNTAX) section of the man page for further details.

Is this something that would be useful to enough people and easy enough to be worth considering implementing?

**Comment by kellyjonbrazil** [comment]
Hi @chriscroome - yeah this looks pretty easy to parse. I'm working on releasing v1.20.0 right now, but this could make it into the next release.

**Comment by kellyjonbrazil** [comment]
I have an initial version of the `postconf` parser ready for testing [here](https://github.com/kellyjonbrazil/jc/blob/dev/jc/parsers/postconf.py). You can copy this file to your [plugins folder](https://github.com/kellyjonbrazil/jc#custom-parsers) to test.

Here is the converted output from the sample you gave. As you can see I needed to make some decisions on how to represent some of the states. These are documented in the schema.
```
[
  {
    "service_name": "smtp",
    "service_type": "inet",
    "private": false,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": -1,
    "process_limit": -1,
    "command": "smtpd",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "pickup",
    "service_type": "unix",
    "private": false,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": 60,
    "process_limit": 1,
    "command": "pickup",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "cleanup",
    "service_type": "unix",
    "private": false,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": -1,
    "process_limit": 0,
    "command": "cleanup",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "qmgr",
    "service_type": "unix",
    "private": false,
    "unprivileged": null,
    "chroot": false,
    "wake_up_time": 300,
    "process_limit": 1,
    "command": "qmgr",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "tlsmgr",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": 1000,
    "process_limit": 1,
    "command": "tlsmgr",
    "no_wake_up_before_first_use": true
  },
  {
    "service_name": "rewrite",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": -1,
    "process_limit": -1,
    "command": "trivial-rewrite",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "bounce",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": -1,
    "process_limit": 0,
    "command": "bounce",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "defer",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": -1,
    "process_limit": 0,
    "command": "bounce",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "trace",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": -1,
    "process_limit": 0,
    "command": "bounce",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "verify",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": -1,
    "process_limit": 1,
    "command": "verify",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "flush",
    "service_type": "unix",
    "private": false,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": 1000,
    "process_limit": 0,
    "command": "flush",
    "no_wake_up_before_first_use": true
  },
  {
    "service_name": "proxymap",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": false,
    "wake_up_time": -1,
    "process_limit": -1,
    "command": "proxymap",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "proxywrite",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": false,
    "wake_up_time": -1,
    "process_limit": 1,
    "command": "proxymap",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "smtp",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": -1,
    "process_limit": -1,
    "command": "smtp",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "relay",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": -1,
    "process_limit": -1,
    "command": "smtp -o syslog_name=postfix/$service_name",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "showq",
    "service_type": "unix",
    "private": false,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": -1,
    "process_limit": -1,
    "command": "showq",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "error",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": -1,
    "process_limit": -1,
    "command": "error",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "retry",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": -1,
    "process_limit": -1,
    "command": "error",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "discard",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": -1,
    "process_limit": -1,
    "command": "discard",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "local",
    "service_type": "unix",
    "private": null,
    "unprivileged": false,
    "chroot": false,
    "wake_up_time": -1,
    "process_limit": -1,
    "command": "local",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "virtual",
    "service_type": "unix",
    "private": null,
    "unprivileged": false,
    "chroot": false,
    "wake_up_time": -1,
    "process_limit": -1,
    "command": "virtual",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "lmtp",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": -1,
    "process_limit": -1,
    "command": "lmtp",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "anvil",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": -1,
    "process_limit": 1,
    "command": "anvil",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "scache",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": -1,
    "process_limit": 1,
    "command": "scache",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "postlog",
    "service_type": "unix-dgram",
    "private": false,
    "unprivileged": null,
    "chroot": false,
    "wake_up_time": -1,
    "process_limit": 1,
    "command": "postlogd",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "maildrop",
    "service_type": "unix",
    "private": null,
    "unprivileged": false,
    "chroot": false,
    "wake_up_time": -1,
    "process_limit": -1,
    "command": "pipe flags=DRXhu user=vmail argv=/usr/bin/maildrop -d ${recipient}",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "uucp",
    "service_type": "unix",
    "private": null,
    "unprivileged": false,
    "chroot": false,
    "wake_up_time": -1,
    "process_limit": -1,
    "command": "pipe flags=Fqhu user=uucp argv=uux -r -n -z -a$sender - $nexthop!rmail ($recipient)",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "ifmail",
    "service_type": "unix",
    "private": null,
    "unprivileged": false,
    "chroot": false,
    "wake_up_time": -1,
    "process_limit": -1,
    "command": "pipe flags=F user=ftn argv=/usr/lib/ifmail/ifmail -r $nexthop ($recipient)",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "bsmtp",
    "service_type": "unix",
    "private": null,
    "unprivileged": false,
    "chroot": false,
    "wake_up_time": -1,
    "process_limit": -1,
    "command": "pipe flags=Fq. user=bsmtp argv=/usr/lib/bsmtp/bsmtp -t$nexthop -f$sender $recipient",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "scalemail-backend",
    "service_type": "unix",
    "private": null,
    "unprivileged": false,
    "chroot": false,
    "wake_up_time": -1,
    "process_limit": 2,
    "command": "pipe flags=R user=scalemail argv=/usr/lib/scalemail/bin/scalemail-store ${nexthop} ${user} ${extension}",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "mailman",
    "service_type": "unix",
    "private": null,
    "unprivileged": false,
    "chroot": false,
    "wake_up_time": -1,
    "process_limit": -1,
    "command": "pipe flags=FRX user=list argv=/usr/lib/mailman/bin/postfix-to-mailman.py ${nexthop} ${user}",
    "no_wake_up_before_first_use": false
  }
]
```

Here is the `raw` output:
```
[
  {
    "service_name": "smtp",
    "service_type": "inet",
    "private": "n",
    "unprivileged": "-",
    "chroot": "y",
    "wake_up_time": "-",
    "process_limit": "-",
    "command": "smtpd"
  },
  {
    "service_name": "pickup",
    "service_type": "unix",
    "private": "n",
    "unprivileged": "-",
    "chroot": "y",
    "wake_up_time": "60",
    "process_limit": "1",
    "command": "pickup"
  },
  {
    "service_name": "cleanup",
    "service_type": "unix",
    "private": "n",
    "unprivileged": "-",
    "chroot": "y",
    "wake_up_time": "-",
    "process_limit": "0",
    "command": "cleanup"
  },
  {
    "service_name": "qmgr",
    "service_type": "unix",
    "private": "n",
    "unprivileged": "-",
    "chroot": "n",
    "wake_up_time": "300",
    "process_limit": "1",
    "command": "qmgr"
  },
  {
    "service_name": "tlsmgr",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "-",
    "chroot": "y",
    "wake_up_time": "1000?",
    "process_limit": "1",
    "command": "tlsmgr"
  },
  {
    "service_name": "rewrite",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "-",
    "chroot": "y",
    "wake_up_time": "-",
    "process_limit": "-",
    "command": "trivial-rewrite"
  },
  {
    "service_name": "bounce",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "-",
    "chroot": "y",
    "wake_up_time": "-",
    "process_limit": "0",
    "command": "bounce"
  },
  {
    "service_name": "defer",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "-",
    "chroot": "y",
    "wake_up_time": "-",
    "process_limit": "0",
    "command": "bounce"
  },
  {
    "service_name": "trace",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "-",
    "chroot": "y",
    "wake_up_time": "-",
    "process_limit": "0",
    "command": "bounce"
  },
  {
    "service_name": "verify",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "-",
    "chroot": "y",
    "wake_up_time": "-",
    "process_limit": "1",
    "command": "verify"
  },
  {
    "service_name": "flush",
    "service_type": "unix",
    "private": "n",
    "unprivileged": "-",
    "chroot": "y",
    "wake_up_time": "1000?",
    "process_limit": "0",
    "command": "flush"
  },
  {
    "service_name": "proxymap",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "-",
    "chroot": "n",
    "wake_up_time": "-",
    "process_limit": "-",
    "command": "proxymap"
  },
  {
    "service_name": "proxywrite",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "-",
    "chroot": "n",
    "wake_up_time": "-",
    "process_limit": "1",
    "command": "proxymap"
  },
  {
    "service_name": "smtp",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "-",
    "chroot": "y",
    "wake_up_time": "-",
    "process_limit": "-",
    "command": "smtp"
  },
  {
    "service_name": "relay",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "-",
    "chroot": "y",
    "wake_up_time": "-",
    "process_limit": "-",
    "command": "smtp -o syslog_name=postfix/$service_name"
  },
  {
    "service_name": "showq",
    "service_type": "unix",
    "private": "n",
    "unprivileged": "-",
    "chroot": "y",
    "wake_up_time": "-",
    "process_limit": "-",
    "command": "showq"
  },
  {
    "service_name": "error",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "-",
    "chroot": "y",
    "wake_up_time": "-",
    "process_limit": "-",
    "command": "error"
  },
  {
    "service_name": "retry",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "-",
    "chroot": "y",
    "wake_up_time": "-",
    "process_limit": "-",
    "command": "error"
  },
  {
    "service_name": "discard",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "-",
    "chroot": "y",
    "wake_up_time": "-",
    "process_limit": "-",
    "command": "discard"
  },
  {
    "service_name": "local",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "n",
    "chroot": "n",
    "wake_up_time": "-",
    "process_limit": "-",
    "command": "local"
  },
  {
    "service_name": "virtual",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "n",
    "chroot": "n",
    "wake_up_time": "-",
    "process_limit": "-",
    "command": "virtual"
  },
  {
    "service_name": "lmtp",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "-",
    "chroot": "y",
    "wake_up_time": "-",
    "process_limit": "-",
    "command": "lmtp"
  },
  {
    "service_name": "anvil",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "-",
    "chroot": "y",
    "wake_up_time": "-",
    "process_limit": "1",
    "command": "anvil"
  },
  {
    "service_name": "scache",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "-",
    "chroot": "y",
    "wake_up_time": "-",
    "process_limit": "1",
    "command": "scache"
  },
  {
    "service_name": "postlog",
    "service_type": "unix-dgram",
    "private": "n",
    "unprivileged": "-",
    "chroot": "n",
    "wake_up_time": "-",
    "process_limit": "1",
    "command": "postlogd"
  },
  {
    "service_name": "maildrop",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "n",
    "chroot": "n",
    "wake_up_time": "-",
    "process_limit": "-",
    "command": "pipe flags=DRXhu user=vmail argv=/usr/bin/maildrop -d ${recipient}"
  },
  {
    "service_name": "uucp",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "n",
    "chroot": "n",
    "wake_up_time": "-",
    "process_limit": "-",
    "command": "pipe flags=Fqhu user=uucp argv=uux -r -n -z -a$sender - $nexthop!rmail ($recipient)"
  },
  {
    "service_name": "ifmail",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "n",
    "chroot": "n",
    "wake_up_time": "-",
    "process_limit": "-",
    "command": "pipe flags=F user=ftn argv=/usr/lib/ifmail/ifmail -r $nexthop ($recipient)"
  },
  {
    "service_name": "bsmtp",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "n",
    "chroot": "n",
    "wake_up_time": "-",
    "process_limit": "-",
    "command": "pipe flags=Fq. user=bsmtp argv=/usr/lib/bsmtp/bsmtp -t$nexthop -f$sender $recipient"
  },
  {
    "service_name": "scalemail-backend",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "n",
    "chroot": "n",
    "wake_up_time": "-",
    "process_limit": "2",
    "command": "pipe flags=R user=scalemail argv=/usr/lib/scalemail/bin/scalemail-store ${nexthop} ${user} ${extension}"
  },
  {
    "service_name": "mailman",
    "service_type": "unix",
    "private": "-",
    "unprivileged": "n",
    "chroot": "n",
    "wake_up_time": "-",
    "process_limit": "-",
    "command": "pipe flags=FRX user=list argv=/usr/lib/mailman/bin/postfix-to-mailman.py ${nexthop} ${user}"
  }
]
```

**Comment by kellyjonbrazil** [comment]
I could change those `-1`'s to `nulls`. Might make more sense.

**Comment by kellyjonbrazil** [comment]
I updated the schema:
```
    [
      {
        "service_name":                     string,
        "service_type":                     string,
        "private":                          boolean/null,  # [0]
        "unprivileged":                     boolean/null,  # [0]
        "chroot":                           boolean/null,  # [0]
        "wake_up_time":                     integer/null,  # [0]
        "no_wake_up_before_first_use":      boolean/null,  # [1]
        "process_limit":                    integer/null,  # [0]
        "command":                          string
      }
    ]

    [0] '-' converted to null/None
    [1] null/None if `wake_up_time` is null/None
```

so the output now looks like this:
```
[
  {
    "service_name": "smtp",
    "service_type": "inet",
    "private": false,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": null,
    "process_limit": null,
    "command": "smtpd",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "pickup",
    "service_type": "unix",
    "private": false,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": 60,
    "process_limit": 1,
    "command": "pickup",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "cleanup",
    "service_type": "unix",
    "private": false,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": null,
    "process_limit": 0,
    "command": "cleanup",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "qmgr",
    "service_type": "unix",
    "private": false,
    "unprivileged": null,
    "chroot": false,
    "wake_up_time": 300,
    "process_limit": 1,
    "command": "qmgr",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "tlsmgr",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": 1000,
    "process_limit": 1,
    "command": "tlsmgr",
    "no_wake_up_before_first_use": true
  },
  {
    "service_name": "rewrite",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": null,
    "process_limit": null,
    "command": "trivial-rewrite",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "bounce",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": null,
    "process_limit": 0,
    "command": "bounce",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "defer",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": null,
    "process_limit": 0,
    "command": "bounce",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "trace",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": null,
    "process_limit": 0,
    "command": "bounce",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "verify",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": null,
    "process_limit": 1,
    "command": "verify",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "flush",
    "service_type": "unix",
    "private": false,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": 1000,
    "process_limit": 0,
    "command": "flush",
    "no_wake_up_before_first_use": true
  },
  {
    "service_name": "proxymap",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": false,
    "wake_up_time": null,
    "process_limit": null,
    "command": "proxymap",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "proxywrite",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": false,
    "wake_up_time": null,
    "process_limit": 1,
    "command": "proxymap",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "smtp",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": null,
    "process_limit": null,
    "command": "smtp",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "relay",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": null,
    "process_limit": null,
    "command": "smtp -o syslog_name=postfix/$service_name",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "showq",
    "service_type": "unix",
    "private": false,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": null,
    "process_limit": null,
    "command": "showq",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "error",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": null,
    "process_limit": null,
    "command": "error",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "retry",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": null,
    "process_limit": null,
    "command": "error",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "discard",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": null,
    "process_limit": null,
    "command": "discard",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "local",
    "service_type": "unix",
    "private": null,
    "unprivileged": false,
    "chroot": false,
    "wake_up_time": null,
    "process_limit": null,
    "command": "local",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "virtual",
    "service_type": "unix",
    "private": null,
    "unprivileged": false,
    "chroot": false,
    "wake_up_time": null,
    "process_limit": null,
    "command": "virtual",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "lmtp",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": null,
    "process_limit": null,
    "command": "lmtp",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "anvil",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": null,
    "process_limit": 1,
    "command": "anvil",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "scache",
    "service_type": "unix",
    "private": null,
    "unprivileged": null,
    "chroot": true,
    "wake_up_time": null,
    "process_limit": 1,
    "command": "scache",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "postlog",
    "service_type": "unix-dgram",
    "private": false,
    "unprivileged": null,
    "chroot": false,
    "wake_up_time": null,
    "process_limit": 1,
    "command": "postlogd",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "maildrop",
    "service_type": "unix",
    "private": null,
    "unprivileged": false,
    "chroot": false,
    "wake_up_time": null,
    "process_limit": null,
    "command": "pipe flags=DRXhu user=vmail argv=/usr/bin/maildrop -d ${recipient}",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "uucp",
    "service_type": "unix",
    "private": null,
    "unprivileged": false,
    "chroot": false,
    "wake_up_time": null,
    "process_limit": null,
    "command": "pipe flags=Fqhu user=uucp argv=uux -r -n -z -a$sender - $nexthop!rmail ($recipient)",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "ifmail",
    "service_type": "unix",
    "private": null,
    "unprivileged": false,
    "chroot": false,
    "wake_up_time": null,
    "process_limit": null,
    "command": "pipe flags=F user=ftn argv=/usr/lib/ifmail/ifmail -r $nexthop ($recipient)",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "bsmtp",
    "service_type": "unix",
    "private": null,
    "unprivileged": false,
    "chroot": false,
    "wake_up_time": null,
    "process_limit": null,
    "command": "pipe flags=Fq. user=bsmtp argv=/usr/lib/bsmtp/bsmtp -t$nexthop -f$sender $recipient",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "scalemail-backend",
    "service_type": "unix",
    "private": null,
    "unprivileged": false,
    "chroot": false,
    "wake_up_time": null,
    "process_limit": 2,
    "command": "pipe flags=R user=scalemail argv=/usr/lib/scalemail/bin/scalemail-store ${nexthop} ${user} ${extension}",
    "no_wake_up_before_first_use": false
  },
  {
    "service_name": "mailman",
    "service_type": "unix",
    "private": null,
    "unprivileged": false,
    "chroot": false,
    "wake_up_time": null,
    "process_limit": null,
    "command": "pipe flags=FRX user=list argv=/usr/lib/mailman/bin/postfix-to-mailman.py ${nexthop} ${user}",
    "no_wake_up_before_first_use": false
  }
]
```

**Comment by chriscroome** [comment]
Amazing work! I'll write a Ansible role to use this and let you know if I have any issues.

**Comment by kellyjonbrazil** [comment]
This is now available in v1.20.1 available on PIP. (binaries and packages will be available soon)
